File: solutions/php/clock/1/Clock.php
<?php
declare(strict_types=1);

class Clock
{
    private int $hours;
    private int $minutes;

    function __construct(int $hours, int $minutes = 0)
    {
        $this->hours = $hours;
        $this->minutes = $minutes;
        $this->normalize();
    }

    function __toString(): string
    {
        return sprintf('%02d:%02d', $this->hours, $this->minutes);
    }

    function add(int $minutes): Clock
    {
        return new Clock($this->hours, $this->minutes + $minutes);
    }

    function sub(int $minutes): Clock
    {
        return new Clock($this->hours, $this->minutes - $minutes);
    }

    private function normalize(): void
    {
        $totalMinutes = $this->hours * 60 + $this->minutes;
        $totalMinutes = (($totalMinutes % (24 * 60)) + 24 * 60) % (24 * 60);
        $this->hours = intdiv($totalMinutes, 60);
        $this->minutes = $totalMinutes % 60;
    }
}

File: solutions/go/clock/1/clock.go
package clock

import "fmt"

type Clock struct {
	hours   int
	minutes int
}

func normalize(h, m int) Clock {
	totalMinutes := h*60 + m
	totalMinutes = ((totalMinutes % (24 * 60)) + 24*60) % (24 * 60)
	hours := totalMinutes / 60
	minutes := totalMinutes % 60
	return Clock{hours, minutes}
}

func New(h, m int) Clock {
	return normalize(h, m)
}

func (c Clock) Add(m int) Clock {
	return normalize(c.hours, c.minutes+m)
}

func (c Clock) Subtract(m int) Clock {
	return normalize(c.hours, c.minutes-m)
}

func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c.hours, c.minutes)
}
